backend/src/federation/validateCommunities.ts

```diff
@@ -0,0 +1,74 @@
+import { IsNull } from '@dbTools/typeorm'
+// eslint-disable-next-line camelcase
+import { requestGetPublicKey as v1_0_requestGetPublicKey } from './client/1_0/FederationClient'
+// eslint-disable-next-line camelcase
+import { requestGetPublicKey as v1_1_requestGetPublicKey } from './client/1_1/FederationClient'
+import { backendLogger as logger } from '@/server/logger'
+import { ApiVersionType } from './enum/apiVersionType'
+
+export async function startValidateCommunities(timerInterval: number): Promise<void> {
+  logger.info(
+    `Federation: startValidateCommunities loop with an interval of ${timerInterval} ms...`,
+  )
+  // better to use setTimeout twice than setInterval once -> see https://javascript.info/settimeout-setinterval
+  setTimeout(function run() {
+    validateCommunities()
+    setTimeout(run, timerInterval)
+  }, timerInterval)
+}
+
+export async function validateCommunities(): Promise<void> {
+  const dbCommunities: DbCommunity[] = await DbCommunity.createQueryBuilder()
+    .where({ verifiedAt: IsNull() })
+    .orWhere('verified_at < last_announced_at')
+    .getMany()
+  /*
+  const dbCommunities: DbCommunity[] = await DbCommunity.getRepository().manager.query(
+    'SELECT * FROM `communities` `Community` WHERE (`Community`.`verified_at` IS NULL OR `Community`.`verified_at` < `Community`.`last_announced_at`)',
+  )
+  */
+  logger.debug(`Federation: found ${dbCommunities.length} dbCommunities`)
+  if (dbCommunities) {
+    dbCommunities.forEach(async function (dbCom) {
+      logger.debug(`Federation: dbCom: ${JSON.stringify(dbCom)}`)
+      const apiValueStrings: string[] = Object.values(ApiVersionType)
+      logger.debug(`suppported ApiVersions=`, apiValueStrings)
+      if (apiValueStrings.includes(dbCom.apiVersion)) {
+        logger.debug(
+          `Federation: validate publicKey for dbCom: ${dbCom.id} with apiVersion=${dbCom.apiVersion}`,
+        )
+        const pubKey = await invokeVersionedRequestGetPublicKey(dbCom)
+        logger.info(
+          `Federation: received publicKey=${pubKey} from endpoint=${dbCom.endPoint}/${dbCom.apiVersion}`,
+        )
+        if (pubKey && pubKey === dbCom.publicKey.toString('hex')) {
+          logger.info(`Federation: matching publicKey:  ${pubKey}`)
+          DbCommunity.update({ id: dbCom.id }, { verifiedAt: new Date() })
+          logger.debug(`Federation: updated dbCom:  ${JSON.stringify(dbCom)}`)
+        }
+        /*
+        else {
+          logger.warn(`Federation: received unknown publicKey -> delete dbCom with id=${dbCom.id} `)
+          DbCommunity.delete({ id: dbCom.id })
+        }
+        */
+      } else {
+        logger.warn(
+          `Federation: dbCom: ${dbCom.id} with unsupported apiVersion=${dbCom.apiVersion}; supported versions=${apiValueStrings}`,
+        )
+      }
+    })
+  }
+}
+
+async function invokeVersionedRequestGetPublicKey(dbCom: DbCommunity): Promise<string | undefined> {
```

Comment:
this must be based on a class -> you must return a whole class with all functions for 1.0, 1.1, ...

Reply:
KISS !!!
on my opinion the simpliest solution to follow:
- a simple "switch case" construct on the predefined enum values
- the amount of values should in future never be more than 4-5 different ones
- each case-value defines which version of client-class in the correct version directory is used

no magic, no unvisible constructs, simple to understand/follow
... and base on YOUR arguments about 2-3 weeks ago...

on my opinion perhaps a part for future optimization

Reply:
I strongly disagree

Reply:
than please give a solution, because i can't follow your requirements...

Reply:
```
class 1.0{
  getPublicKey(){..}
  func2(){...}
}
```

```
class 2.0{
  getPublicKey(){..}
  func2(){...}
  func3(){...}
}
```

getApiHandler(version){
  switch(version){
    case 1.0: return new 1.0()
    case 2.0: ...
   }
}

Reply:
as descided...